includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1171,13 +1171,21 @@ public function maybe_filter_gateway_title( $title, $id ) {
 	 * @return void
 	 */
 	public function set_payment_method_title_for_email( $order ) {
+		$payment_gateway = wc_get_payment_gateway_by_order( $order );
+
+		if ( self::GATEWAY_ID !== $payment_gateway->id || ! WC_Payments_Features::is_upe_legacy_enabled() ) {
```

Comment:
It looks like [`wc_get_payment_gateway_by_order` could return `false`](https://github.com/woocommerce/woocommerce/blob/bf80b7a5341be3adcfeae6322408789b37995f71/plugins/woocommerce/includes/wc-order-functions.php#L819), making `$payment_gateway->id` throw an error.

Before checking for the gateway's attribute, could we ensure that `$payment_gateway` is not `empty`?